### ISSUE TITLE
Use of trademarked icon and name

### ISSUE THREAD

**Issue by bbhtt** [flathub/com.github.IsmaelMartinez.teams_for_linux#110]
@jijojosephk Hello we are tracking a bunch of applications with trademark violations and this is on the list.

The current icon is exactly a copy of the offcial Microsoft teams icon and also uses the "Teams" as part of the application name.

Please try to use your own name and icon for the application to avoid trademark issues and confusing people. I think 2 months is a reasonable time period for us to wait for the icon and name to be changed.

You can find some resources for icon guidelines in https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#app-icon and name in https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#app-name

And the guidelines on trademarks https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines#no-trademark-violations

**Comment by jijojosephk** [comment]
@IsmaelMartinez

**Comment by Mikaela** [comment]
Am I reading the link correctly that a quick fix would be renaming to `teams-web-wrapper-for-linux`?

**Comment by bbhtt** [comment]
Unfortunately no.

The icon needs to be changed.

The name should be either distinct for example we have a whatsapp client called "Youp" and "Whatsie", or something like the example in the docs will be fine "Tally for Plausible".

**Comment by IsmaelMartinez** [comment]
Hi @bbhtt , Thanks for letting us know. 

I believe Microsoft trademark is with "Microsoft Teams" not with the word "Teams" (as from their https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RWArzw&ved=2ahUKEwjU85_p3LOGAxX9VEEAHfOvBWUQFnoECBsQAQ&usg=AOvVaw3_T38fLas0oag108pzdEeH)

For that reason, the trademark violation would be if we use their whole name (like "Microsoft Teams for Linux"). I would appreciate some clarification on that aspect. 

Regarding the logo, yes... I will see how we can change it. Thanks for reporting this and I do await for clarification on if Microsoft do hold the word teams as trademark.

**Comment by bbhtt** [comment]
In the context of an unofficial client "Teams for Linux" is a bit problematic.

I suggest you name it something like "___ for Teams" going by "Youp for Whatsapp" or "Tally for Plausible"

**Comment by IsmaelMartinez** [comment]
I am not good at names nor icon's so asked for some help from the community. 

As my understanding that the word teams is not trademark, I will leave that change to later but would try to think of ideas. 

Would the following names pass the trademark and follow your guidelines?
* TeamsPortal
* TeamsWrapper

Thanks again

**Comment by bbhtt** [comment]
They would not cause issues with trademark, but I don't know if they'll pass quality guidelines. I don't moderate that. Personally, the first one is ok, the second is seems bad.

I asked around for some suggestions from people, posting what I got:

```
Colligate
Workforce
Commune
Groups
```

> TeamsPortal

"Portal for Teams" sounds better to me.

**Comment by nursoda** [comment]
I dislike "Portal for Teams" since I associate "portal" rather with the name of a specific website than with an Electron wrapper app. I think of T4L as "application specific browser app with enhanced features". Personally, I'd even go for "T4L" as principal name.

Other ideas: "WebApp for Teams", "Teams on Electron"

BTW, is T4L restricted to Linux or should it run on any platform supporting Electron? The [app icon I sketched](https://github.com/IsmaelMartinez/teams-for-linux/discussions/1277#discussioncomment-9688136) is a plain representation of T4L… so the Name should be decided upon first. Once it's fixed, the logo should try to associate to the name.

**Comment by bbhtt** [comment]
https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines

**Comment by IsmaelMartinez** [comment]
Hi @bbhtt ,

Ok, the names that have made the initial cut are:

* Portal for Teams 
* WebApp for Teams
* Wrapper for Teams
* Teams on Electron 

Can you confirm those names are ok? I am running a open voting at the moment. 

We want to keep the word "teams" as otherwise people will not find it, and also, because the package will still be teams-for-linux for all other packaging systems.

Once the name is confirmed, then we will look at the logo in more detail. Thanks

**Comment by bbhtt** [comment]
Yes seems fine.

**Comment by IsmaelMartinez** [comment]
The name of the app should now be "Portal for Teams". Not sure if we will make it into the 2 months for the logo, because of holidays, and this is not a paid job for me, but we will try. Hope this is understandable

**Comment by bbhtt** [comment]
Yes it's fine, take your time.

**Comment by bbhtt** [comment]
Any update regarding the logo?

**Comment by IsmaelMartinez** [comment]
Working on it but holidays are slowing things down. We should have something in August. Apologies for the delay

**Comment by IsmaelMartinez** [comment]
Apologies for the delay. Are you happy with the following logo (in terms of flathub)? Thanks
 
![512x512](https://github.com/user-attachments/assets/1c5d74dc-e3fe-4500-84d4-cd7e3d4a16bb)

**Comment by bbhtt** [comment]
Seems fine.

**Comment by IsmaelMartinez** [comment]
I just done a pre-release, if all goes well, that would be promoted and then we can close this case. Apologies for the delay but this sort of changes can be quite controversial among users, so need to be taken the time and care needed (also, I only do this on my free time, so things do take longer)

**Comment by IsmaelMartinez** [comment]
Hi @bbhtt , I did try to [release this](https://github.com/flathub/com.github.IsmaelMartinez.teams_for_linux/pull/126) but the build is failing with:

"** (process:3229994): ERROR **: 12:06:43.890: Cannot get the default GSettingsSchemaSource - is the gsettings-desktop-schemas package installed?"

<img width="1229" alt="image" src="https://github.com/user-attachments/assets/defa0912-2e3e-4684-9cee-f4728619433f">

I see similar (but not the same) issue in https://github.com/flathub/flathub/pull/4820 and https://github.com/flathub/org.electronjs.Electron2.BaseApp/issues/58 (both that you replied). 

I don't really know flathub (this use to be managed by someone else) so can you give me a hint? I don't see any other errors in the logs (no icon size, etc) but I might be looking at the wrong place.

Thanks in advance!

**Comment by bbhtt** [comment]
You can ignore those, the error is in the last lines of the log. Something is wrong with your desktop file

```
Renaming teams-for-linux.desktop to com.github.IsmaelMartinez.teams_for_linux.desktop
Error: Error on line 27: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
```

**Comment by IsmaelMartinez** [comment]
Good catch, I am fairly sure is this & is creating the issue. 

![image](https://github.com/user-attachments/assets/27b759da-94af-4abe-904e-7ea1f5e84ddd)

Building in my project now,  hopefully the bot will pick up the changes.  

Thanks!

**Comment by bbhtt** [comment]
Yea that should be &amp;

**Comment by nursoda** [comment]
So, this issue may be closed then?

**Comment by IsmaelMartinez** [comment]
waiting for a PR to get merged and this should be released as 1.10.1 . Then we can close it but it is not live yet as 1.10.0 has that & issue, but I don't like releases to not be atomic (re-releasing an already released release ain't a good pattern). Hopefully tomorrow.

**Comment by IsmaelMartinez** [comment]
I am closing this now as the icon and name have changed. Please reopen with more info if they are still violating the guidelines. Thanks for your assistance 

<img width="523" alt="image" src="https://github.com/user-attachments/assets/78a805c2-ed8d-4686-b9b7-4027b951fe7a">
